src/EnergyPlus/UnitarySystem.cc

```diff
@@ -1088,28 +1088,52 @@ namespace UnitarySystems {
                         this->checkNodeSetPoint(state, AirLoopNum, ControlNode, CoolingCoil, OAUCoilOutTemp);
+                    } else if (this->m_ControlType == UnitarySysCtrlType::Setpoint) {
+                        ShowSevereError(state, format("{}: Missing cooling set point in {} = {}", routineName, this->UnitType, this->Name));
+                        ShowContinueError(state,
+                                          format("...Setpoint is required at system air outlet node = {} or cooling coil air outlet node = {}",
+                                                 state.dataLoopNodes->NodeID(this->AirOutNode),
+                                                 state.dataLoopNodes->NodeID(this->CoolCoilOutletNodeNum)));
+                        setPointMissing = true;
                     }
                 }
 
                 if (this->m_HeatCoilExists) {
                     int ControlNode = this->m_SystemHeatControlNodeNum;
                     if (ControlNode > 0) {
                         this->checkNodeSetPoint(state, AirLoopNum, ControlNode, HeatingCoil, OAUCoilOutTemp);
+                    } else if (this->m_ControlType == UnitarySysCtrlType::Setpoint) {
+                        ShowSevereError(state, format("{}: Missing heating set point in {} = {}", routineName, this->UnitType, this->Name));
+                        ShowContinueError(state,
+                                          format("...Setpoint is required at system air outlet node = {} or heating coil air outlet node = {}",
+                                                 state.dataLoopNodes->NodeID(this->AirOutNode),
+                                                 state.dataLoopNodes->NodeID(this->HeatCoilOutletNodeNum)));
+                        setPointMissing = true;
                     }
                 }
 
                 if (this->m_SuppCoilExists) {
                     int ControlNode = this->m_SuppHeatControlNodeNum;
                     if (ControlNode > 0) {
                         this->checkNodeSetPoint(state, AirLoopNum, ControlNode, SuppHeatCoil, OAUCoilOutTemp);
+                    } else if (this->m_ControlType == UnitarySysCtrlType::Setpoint) {
+                        ShowSevereError(state,
+                                        format("{}: Missing supplemental heater set point in {} = {}", routineName, this->UnitType, this->Name));
+                        ShowContinueError(
+                            state,
+                            format("...Setpoint is required at system air outlet node = {} or supplemental heating coil air outlet node = {}",
+                                   state.dataLoopNodes->NodeID(this->AirOutNode),
+                                   state.dataLoopNodes->NodeID(this->m_SuppCoilAirOutletNode)));
+                        setPointMissing = true;
                     }
                 }
 
+                if (setPointMissing) ShowFatalError(state, "Previous errors cause termination.");
```

Comment:
If I'm following correctly, if the control node is zero, then we get a fatal error here, but if `checkNodeSetPoint` throws error, there is no fatal?

Reply:
That is what I see also, no fatal for set point errors on humidity control. Took your good suggestion and moved new code to checknodeSetPoint. Also added unit testing. And begin moving away from m_* variable names.